### PR TITLE
feature: Enable runner targeting by labels specified in runner profile

### PR DESCRIPTION
**Pull request by xiaolin-ninja**
Another piece of the targetable runners puzzle. 

This PR adds tests for job successfully queued and targeted, and CLI changes to allow creation of runner profiles with labels.
It also makes an important choice to allow definition of a runner profiles that targets a runner that does not exist yet. (Previous iteration of targetable runners did not allow this.)

```
❯ waypoint-dev runner profile set -name=efgh -target-runner-label=env=test -target-runner-label=region=testland -plugin-type=kubernetes
✓ Runner profile created
```
```
❯ waypoint-dev runner profile set -name=efgh -target-runner-label=env=test -target-runner-id=test -plugin-type=kubernetes 
✓ Runner profile updated
Both -target-runner-id and -target-runner-label detected, only one can be set at a time. ID takes priority.
```
```
❯ waypoint-dev runner profile list
Runner profiles
NAME | PLUGIN TYPE |            OCI URL            |               TARGET RUNNER                | DEFAULT  
-----+-------------+-------------------------------+--------------------------------------------+----------
1234 | kubernetes  | hashicorp/waypoint-odr:latest | *                                          | yes
abcd | kubernetes  | hashicorp/waypoint-odr:latest | test                                       |
efgh | kubernetes  | hashicorp/waypoint-odr:latest | labels: {"env":"test","region":"testland"} |    
```
```
❯ waypoint-dev runner profile inspect efgh

» Runner profile:
                   Name: efgh
                     ID: 01FYZBPNEWXEP00W5ZWN78XVD9
                Default: false
                OCI URL: hashicorp/waypoint-odr:latest
            Plugin Type: kubernetes
   Target Runner Labels: map[env:test region:testland]
  Environment Variables: map[]
```

### CODE REVIEW

**Comment by xiaolin-ninja** [comment]
Connected to: 
https://github.com/hashicorp/waypoint/pull/3133
https://github.com/hashicorp/waypoint/pull/3132

File: internal/cli/runner_profile_inspect.go [hunk 1]
```diff
@@ -106,6 +109,9 @@ func (c *RunnerProfileInspectCommand) Run(args []string) int {
 		{
 			Name: "Target Runner ID", Value: targetRunner,
 		},
+		{
+			Name: "Target Runner Labels", Value: targetRunnerLabels,
+		},
```
Comment (mitchellh) [inline internal/cli/runner_profile_inspect.go:114]: Not a big deal, but a target runner can have only one of any, id, or lables, so should this be one field that is like `Target:` instead that follows the format of `list` or something?

File: internal/cli/runner_profile_list.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package cli
 
 import (
+	"encoding/json"
```
Comment (mitchellh) [inline internal/cli/runner_profile_list.go:4]: Nitpick, across the project we typically do the pattern below, so we should have a newline between this and the next imports. Lots of tools automatically insert and break this once in awhile, not a big deal usually just manually clean it up.

```
import (
  stdlib

  external

  internal
)
```

File: internal/cli/runner_profile_set.go [hunk 3]
```diff
@@ -269,7 +278,14 @@ func (c *RunnerProfileSetCommand) Flags() *flag.Sets {
...
 			Default: "",
-			Usage:   "ID of the remote runner to target for the profile.",
+			Usage:   "ID of the runner to target for this remote runner profile.",
+		})
+
+		f.StringMapVar(&flag.StringMapVar{
+			Name:   "target-runner-label",
+			Target: &c.flagTargetRunnerLabels,
+			Usage: "Labels on the runner to target for this remote runner profile. " +
+				"e.g. `-target-runner-labels=k=v`. Can be specified multiple times.",
```
Comment (mitchellh) [inline internal/cli/runner_profile_set.go:288]: plural to singular here

**Review by mitchellh — APPROVED** [review]
Just a few nitpicks. Approved!

File: internal/cli/runner_profile_set.go [hunk 4]
```diff
@@ -269,7 +278,14 @@ func (c *RunnerProfileSetCommand) Flags() *flag.Sets {
...
 			Target:  &c.flagTargetRunnerId,
 			Default: "",
-			Usage:   "ID of the remote runner to target for the profile.",
+			Usage:   "ID of the runner to target for this remote runner profile.",
+		})
+
+		f.StringMapVar(&flag.StringMapVar{
+			Name:   "target-runner-label",
+			Target: &c.flagTargetRunnerLabels,
+			Usage: "Labels on the runner to target for this remote runner profile. " +
```
Comment (briancain) [inline internal/cli/runner_profile_set.go:287]: Maybe this could use an additional sentence on what labels for targeting is used for?

**Review by briancain — APPROVED** [review]
Looks pretty good to me! 👍🏻 I'll let Mitchell chime in since he was following the previous puzzle pieces, but overall makes sense to me

File: internal/cli/runner_profile_inspect.go (hunk 1 above)
Comment (xiaolin-ninja) [reply internal/cli/runner_profile_inspect.go:114]: Hmm that's a good point actually, right now it's not hard-coded to restrict only one of ID or Labels, but the CLI presents it as such, and ID always takes precedence over labels.

**Pull request merged**